backstory/src/main/java/com/google/sps/story/StoryManagerImpl.java

```diff
@@ -144,4 +153,28 @@ private String makeRequestBody(String prefix, int maxLength, Double temperature)
     String convertedMap = gson.toJson(requestMap);
     return convertedMap;
   }
+
+  /**
+   * Cycles to next serviceUrl to a backup url.
+   * @return False if called on last URL.
+   */
+  public boolean cycleUrl() {
+    boolean isLast = true;
+    if (selectedUrlIndex == serviceUrls.length - 1) {
+      isLast = false;
+    }
+    if (selectedUrlIndex < serviceUrls.length) {
+      selectedUrlIndex++;
+    } else {
+      selectedUrlIndex = 0;
+    }
+    return isLast;
+  }
+
+  /**
+   * Sets prompt for text generation.
```

Comment:
You should probably have a @param tag here.

```suggestion
   * Sets prompt for text generation.
   *
   * @param prefix the prefix to act as a prompt 
```